### ISSUE TITLE
Difference in VRAM utilization (single vs separate python script) and (ubuntu vs Windows)

### ISSUE THREAD

**Issue by kmvolv** [ultralytics/yolov5#12246]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

## Question 1

The VRAM utilization is lower when multiple models are loaded in the same python script but when the same models are loaded in different python scripts it is doubled. 

For example : Let's assume loading yolov5s model take 200MB of VRAM, if I load 2 yolov5s in 2 different scripts it will be 400MB but if i load the same models in a single python script it would be ~240 MB, is something being shared in VRAM that is causing this difference?

Actual results when models are loaded in same scripts : 
![image](https://github.com/ultralytics/yolov5/assets/118155144/0551c731-aa41-4b08-8ef5-4e66e6a36302)

We can see that i can load 5 models of yolov5x with a batch size (BS) of 1 in around 2.9GB of VRAM, but if i load the same 5 models in 5 different scripts with BS of 1 my VRAM would be around 2.9*5 = 14.5GB , which is a huge difference.

How and why is this happening? does loading multiple models in same script cause any issues during inference?

Code to replicate results : 
```python
import torch
import cv2
import time


class InferModel:
    def __init__(self, name, num_models, size, batch):
        self.num_models = num_models
        self.models = []
        for _ in range(num_models):
            self.models.append(torch.hub.load('ultralytics/yolov5', name))
        self.orig_pic = cv2.imread("wave.png")
        self.orig_pic = cv2.resize(self.orig_pic, size[::-1])
        self.pics = [self.orig_pic]*batch
    def warmup(self):
        for i in range(self.num_models):
            mymodel = self.models[i]
            mymodel(self.orig_pic)
            print(f"============ Warmed up model {i+1} ============")


    def infer(self):
        for i in range(self.num_models):
            mymodel = self.models[i]
            result = mymodel(self.pics)
            print(f"======== This is result {i+1} =========\n\n")
            print(result)
            print("\n\n======================================")


obj = InferModel("yolov5x", 2, (1920,1080), 1)
obj.warmup()
obj.infer()


fin = input("Press enter to exit")
```

## Question 2
The utilization is significantly higher in Windows when compared to Linux using the same hardware and script. 
Why does this happen?

![image](https://github.com/ultralytics/yolov5/assets/72385838/7e3d08da-46e3-4839-9b40-9daffbd9483d)

win version details WIN 10 PRO
Python 3.8.5
Torch 2.1.0+cu121
CUDA  V12.2.140
Driver  537.42

Ubuntu version details : 20.04 LTS
Python 3.8.12
Torch 2.0.1+cu117
CUDA v11.7.64
Driver 535.104.12

h/w details:
CPU : https://ark.intel.com/content/www/us/en/ark/products/199332/intel-core-i910900k-processor-20m-cache-up-to-5-30-ghz.html
GPU : https://www.techpowerup.com/vgabios/225687/zotac-rtx3090-24576-200901
RAM : 32 gb

**Comment by glenn-jocher** [comment]
@kmvolv the difference in VRAM utilization between loading multiple models in the same Python script and loading them in separate scripts can be explained by memory allocation behavior. When loading models in separate scripts, each script creates its own memory space, resulting in duplicated VRAM usage. On the other hand, when models are loaded in the same script, memory may be shared, leading to more efficient memory utilization.

Regarding your concern about loading multiple models in the same script causing issues during inference, I can assure you that YOLOv5 is designed to handle multiple models efficiently. However, keep in mind that the total VRAM usage depends on the size and complexity of the models as well as the size of the input images.

As for the discrepancy in VRAM utilization between Windows and Linux, it is common to observe variations due to the differences in OS drivers, memory management, and other system-specific factors. These variations can impact the reported VRAM utilization. However, the difference in hardware utilization should not affect the performance or accuracy of the inference.

Please note that the provided code does not have any issues and effectively demonstrates the difference in VRAM utilization. If you have any further questions, feel free to ask!

**Comment by kmvolv** [comment]
Thank you for the prompt reply, both of my doubts have been clarified and I have no further questions.

**Comment by glenn-jocher** [comment]
@kmvolv i'm glad that I could help clarify your doubts. If you have any more questions in the future, feel free to ask. Have a great day!
